specification/purview/data-plane/Azure.Analytics.Purview.Scanning/stable/2023-09-01/scanningService.json

```diff
@@ -0,0 +1,10292 @@
+            "AdlsGen2Msi",
+            "AmazonAccountCredential",
+            "AmazonS3Credential",
+            "AmazonS3RoleARN",
+            "AmazonSqlCredential",
+            "AzureCosmosDbCredential",
+            "AzureDataExplorerCredential",
+            "AzureDataExplorerMsi",
+            "AzureFileServiceCredential",
+            "AzureSqlDatabaseCredential",
+            "AzureSqlDatabaseMsi",
+            "AzureSqlDatabaseNativeAuth",
+            "ArcEnabledSqlServerCredential",
+            "ArcEnabledSqlServerNativeAuth",
+            "AmazonPostgreSqlCredential",
+            "AzurePostgreSqlCredential",
+            "DatabricksCredential",
+            "SqlServerDatabaseCredential",
+            "AzureSqlDatabaseManagedInstanceCredential",
+            "AzureSqlDatabaseManagedInstanceMsi",
+            "AzureSqlDatabaseManagedInstanceNativeAuth",
+            "AzureSqlDataWarehouseCredential",
+            "AzureSqlDataWarehouseMsi",
+            "AzureMySqlCredential",
+            "HdfsCredential",
+            "AzureStorageCredential",
+            "AzureStorageMsi",
+            "TeradataTeradataCredential",
+            "TeradataTeradataUserPass",
+            "TeradataUserPass",
+            "OracleOracleCredential",
+            "OracleOracleUserPass",
+            "DatabricksHmsCredential",
+            "SapS4HanaSapS4HanaCredential",
+            "SapS4HanaSapS4HanaUserPass",
+            "SapEccSapEccCredential",
+            "SapEccSapEccUserPass",
+            "SnowflakeCredential",
+            "PowerBICredential",
+            "PowerBIDelegated",
+            "PowerBIFpa",
+            "PowerBIMsi",
+            "FabricCredential",
+            "FabricDelegated",
+            "FabricFpa",
+            "FabricMsi",
+            "TridentCredential",
+            "TridentDelegated",
+            "TridentMsi",
+            "DataverseCredential",
+            "DataverseMsi"
+          ],
+          "type": "string",
+          "description": "The properties of scan, vary by datasource type.",
+          "x-ms-enum": {
+            "name": "ScanAuthorizationType",
+            "modelAsString": true
+          }
+        },
+        "scanResults": {
+          "description": "Scan results of this scan.",
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/ScanResult"
+          },
+          "readOnly": true
+        },
+        "scanResultsWithIngestion": {
+          "type": "array",
+          "description": "Scan results with ingestion of this scan.",
+          "items": {
+            "$ref": "#/definitions/ScanResultWithIngestion"
+          },
+          "readOnly": true
+        },
+        "scanId": {
+          "type": "string",
+          "description": "The scan identifier."
+        },
+        "dataSourceIdentifier": {
+          "allOf": [
+            {
+              "$ref": "#/definitions/DataSourceIdentifier"
+            }
+          ]
```

Comment:
Similarly, avoid meaningless allOf. It seems also confuse msdoc https://review.learn.microsoft.com/en-us/rest/api/documentation-preview/scans/create-or-replace?view=azure-rest-preview&branch=result_fb0bc5b2-8509-43d6-a090-fbeed840ff68&tabs=HTTP#datasourceidentifier

Reply:
I didn't expect all of them handled, but prioritize those user likely to use.

Reply:
Just removed this allOf